meshio/vtk_io.py

```diff
@@ -472,16 +506,27 @@ def _write_cells(f, cells, write_binary):
     f.write("CELL_TYPES {}\n".format(total_num_cells).encode("utf-8"))
     if write_binary:
         for key in cells:
-            d = numpy.full(len(cells[key]), meshio_to_vtk_type[key]).astype(
-                numpy.dtype(">i4")
-            )
+            if "polygon" not in key:
+                d = numpy.full(len(cells[key]), meshio_to_vtk_type[key]).astype(
+                    numpy.dtype(">i4")
+                )
+            else:
+                d = numpy.full(len(cells[key]), meshio_to_vtk_type["polygon"]).astype(
```

Comment:
This looks like it's not going to work if both `polygon` and other cells are in the mesh.

Reply:
why? It seems to work with my first tests

Reply:
If the `"polygon"` key _is_ in `cells` (along with other keys), then you always pick `meshio_to_vtk_type["polygon"]`. That's the part that looks suspicious to me.

Reply:
Nope, there is loop over the keys and the test is made inside the loop

Reply:
Ah, same misunderstanding as above. When I read `in` I think lists. Please also replace by `key[:7] == "polygon"`.